midpoint/install/system-requirements/index.adoc

```diff
@@ -223,93 +230,123 @@ Refer to the xref:/midpoint/release/[midPoint Releases] documentation for softwa
 
-Identity engineers need VPN access to all nodes (node 1, node2: SSH & Tunnel, and also HTTP/S access to the local Tomcat that can be tunneled).
-Node jobs are synchronized over JMX.
+A separate SSH access for troubleshooting purposes is advised, so that you have a chance to examine the situation in case the VPN or the load balancer fail.
 
-Each midPoint node needs to have access to the SMTP server (if notifications are required), shared HA DB Repository, and all source and target systems to have full HA support.
-When a node is down, other nodes need to be able to replace all of its jobs.
+==== Resources
 
-End users and identity engineers use midPoint Web UI over a load balancer (HTTPS).
+Then there are the resources.
+There are many different communication protocols the resources may use.
+The schema above shows just a couple of the most common ones.
 
-You can check that connections to all source or target systems are working using the following tools: `ping`, `telnet`, and `wget`.
-You can use them from the midPoint server and also from the workstation (once tunnels have been configured).
+You may have an HR system which is only able to give midPoint CSV files, there may be a more complex system, data of which you can access using SQL, you may need to connect an Active Directory, and so on.
+With some resources, such as xref:/connectors/connectors/com.evolveum.polygon.connector.sap.SapConnector/[SAP and JCo], you need to enable API on the target resource, open your firewall on servers where the target system is running, and create an account with appropriate permissions to manage identities.
+In certain cases, you may have to deploy a <<connector_servers_sizing,connector server>> to access some special resources.
+The list of options is endless and very much depends on what exactly you need to manage with midPoint.
+
+=== Multi-Node Deployment
+
+The situation gets a bit more complex when you deploy midPoint on two or more nodes:
+
+//image::environment-schema-HA.png[]
+image::midpoint-environment-schema-HA-two-nodes.svg[title="Deployment of midPoint with two nodes, load balancer, VPN, and a few example resources"]
+
+==== Work Distribution
+
+In multi-node deployment, the job distribution among the nodes is handled centrally.
+
+The repository DB keeps track of the task states, i.e., which are to be done, in progress, or done.
+This helps to ensure that no single task is processed by two nodes concurrently.
+The Quartz library then serves as a smart timer, creating worker tasks which it gives to nodes for processing.
+The available tasks are picked by the nodes on the first-come-first-served basis, which ensures reasonably even task distribution.
+
+Refer to xref:/midpoint/reference/tasks/task-manager/[] for details on task handling in midPoint.
+
+==== Communication Among Nodes
+
+Primarily, the nodes communicate with the central Quartz library rather than among themselves,
```

Comment:
they communicates with central repository, not quartz. Quartz is just a smart scheduler.